### DISCUSSION TITLE
How to test maplibre styles loading?

### DISCUSSION THREAD

**Discussion by guimochila** [maplibre/maplibre-gl-js#2193] [Q&A] [answered]
Hi everyone,

We are implementing tests to our application, however when testing maplibre load/reload styles we are facing lots of instability in our tests, one of our engineers wrote:

> the problem is that you apply a style change and need to verify that it was applied correctly on map, but you don’t know when exactly this style changes are done. Sometimes we wait for a specific number of features to be queried, other times we just set arbitrary timeout before we assert

How are can make sure that maplibre load and applied all changes to the map before we assert something?

Thanks in advance.

**Comment by IvanSanchez** [comment, marked as answer]
My guess would be to listen to `data` events (see https://maplibre.org/maplibre-gl-js-docs/api/map/#map.event:data and https://maplibre.org/maplibre-gl-js-docs/api/events/#mapdataevent), looking for a `dataType` property with a value of `style`.

**Reply by wipfli** [reply]
Or maybe this one? https://maplibre.org/maplibre-gl-js-docs/api/map/#map.event:render

**Reply by HarelM** [reply]
There's also the `idle` event.

**Reply by guimochila** [reply]
Thanks all, I will have a look at them, 'idle' seems to be an interesting approach.
